File: infrastructure/web/dtos.go
package web

type errorResponse struct {
	Code        string `json:"ERR_CODE"`
	Description string `json:"Description"`
}

type createResponse struct {
	Alias       string `json:"alias"`
	Original    string `json:"original"`
	Shortened   string `json:"shortened"`
	ElapsedTime string `json:"elapsed_time"`
}

File: client/src/dtos.ts
export type ShortenerError = {
  ERR_CODE: string
  Description: string
}

export type UrlInfo = {
  alias: string
  original: string
  shortened: string
  elapsedTime: string
}

export type MostVisitedUrl = {
  url: string
  short_url: string
  visits: number
}
